### PR TITLE
sessionctx: add tidb_enable_distribute_reorg session variable

### LINKED ISSUE #37120: Global variable controls the switch of distribution reorg

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #37120

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: .github/workflows/license-checker.yml [hunk 1]
```diff
@@ -3,10 +3,10 @@ name: License checker
 on:
   push:
     branches:
-      - master
+      - distribute-reorg
   pull_request:
     branches:
-      - master
+      - distribute-reorg
```
Comment (zimulala) [inline .github/workflows/license-checker.yml:9]: Sorry. Recently, someone raised the issue of the naming specification of the feature branch, which I changed its name to feature/distribute-reorg.

File: sessionctx/variable/tidb_vars.go [hunk 2]
```diff
@@ -776,6 +776,8 @@ const (
 	TiDBMaxAutoAnalyzeTime = "tidb_max_auto_analyze_time"
 	// TiDBEnableConcurrentDDL indicates whether to enable the new DDL framework.
 	TiDBEnableConcurrentDDL = "tidb_enable_concurrent_ddl"
+	// TiDBEnableDistributeReorg indicates whether to enable the new Reorg framework.
```
Comment (zimulala) [inline sessionctx/variable/tidb_vars.go:779]: Maybe `TiDBDDLEnableDistributeReorg` is better? It is more unified with other DDL variables.

File: sessionctx/variable/tidb_vars.go [hunk 3]
```diff
@@ -1052,6 +1055,7 @@ var (
 	PreparedPlanCacheSize             = atomic.NewUint64(DefTiDBPrepPlanCacheSize)
 	PreparedPlanCacheMemoryGuardRatio = atomic.NewFloat64(DefTiDBPrepPlanCacheMemoryGuardRatio)
 	EnableConcurrentDDL               = atomic.NewBool(DefTiDBEnableConcurrentDDL)
+	EnableDistributeReorg             = atomic.NewBool(DefTiDBEnableDistributeReorg)
```
Comment (zimulala) [inline sessionctx/variable/tidb_vars.go:1058]: Ditto.

File: sessionctx/variable/sysvar.go [hunk 4]
```diff
@@ -828,6 +828,14 @@ var defaultSysVars = []*SysVar{
 	}, GetGlobal: func(s *SessionVars) (string, error) {
 		return BoolToOnOff(EnableConcurrentDDL.Load()), nil
 	}},
+	{Scope: ScopeGlobal, Name: TiDBEnableDistributeReorg, Value: BoolToOnOff(DefTiDBEnableDistributeReorg), Type: TypeBool, SetGlobal: func(s *SessionVars, val string) error {
```
Comment (zimulala) [inline sessionctx/variable/sysvar.go:831]: Please add tests for this variable.

**Comment by hawkingrei** [comment]
@Defined2014 You should update CI script to use Bazel image on your pipeline

**Comment by Defined2014** [comment]
> @Defined2014 You should update CI script to use Bazel image on your pipeline

cc @zimulala PTAL

File: .github/workflows/license-checker.yml [hunk 5]
```diff
@@ -3,10 +3,10 @@ name: License checker
 on:
   push:
     branches:
-      - master
+      - feature/distribute-reorg
   pull_request:
     branches:
-      - master
+      - feature/distribute-reorg
```
Comment (morgo) [inline .github/workflows/license-checker.yml:9]: I think this should be reverted? Otherwise LGTM

Comment (Defined2014) [reply .github/workflows/license-checker.yml:9]: It will merge to feature branch. So we should modify it to enable license checker.

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request closed without merging**